modules/internal/OwnedObject.chpl

```diff
@@ -129,12 +129,16 @@ module OwnedObject {
 
        :arg p: the class instance to manage. Must be of class type.
      */
-    proc init(p) {
-      this.t = _to_borrowed(p.type);
+    proc init(p:borrowed) {
+      this.t = p.type;
 
-      if !isClass(p) then
-        compilerError("Owned only works with classes");
+      this.p = p;
+    }
 
+    proc init(p:?T) where isClass(T) == false && isSubtype(T, _owned) == false  &&
+                    isIterator(p) == false {
+      compilerError("Owned only works with classes");
+      this.t = T;
```

Comment:
Why is there code after the compilerError?  Shouldn't it never be executed?

Reply:
It won't, but the compiler doesn't really do any kind of unreachable code analysis today, so the regular initializer warnings/errors would apply if I didn't initialize the fields.